image/api_selector.go

```diff
@@ -230,7 +236,15 @@ func (as *APISelector) buildCandidateTags(params *Params) []*tagSet {
 		sort.Strings(ts.Tags)
 	}
 
-	return result
+	for _, ts := range result {
+		for _, tag := range ts.Tags {
+			if strings.Contains(tag, ",") {
+				return result, workerErrors.NewWrappedJobAbortError(errors.Errorf("tag %v contained \",\", which is not supported by job-board -- check .travis.yml for trailing comma", tag))
```

Comment:
I think this specific error should probably be caught in Gatekeeper for a nicer user-facing error, but I also don't think there should be a case where invalid input data into worker can cause an infinite requeue loop, so some form of error handling should be done in worker.